src/nix/eval.cc

```diff
@@ -11,11 +11,13 @@
 
 using namespace nix;
 
+namespace nix::fs { using namespace std::filesystem; }
```

Comment:
I hadn't noticed this before, but there are a few of these `nix::fs` namespace declarations in src/nix now. I don't really understand what that's for? Just use `std::filesystem` so people don't have to wonder what `nix::fs` is.

Reply:
That's because of https://github.com/NixOS/nix/blob/26c3fc11eada3fa7df0284190095868a947fefe2/src/libutil/file-system.hh#L129-L144

I hope that with more `std::filesystem`, most of `file-system.hh` goes away. However, there are a few things that aren't in there they we need, like this `symlink_exists`.

I put it in an `fs` namespace with the idea that if C++ files (not headers!) use `fs::` as a shorthand for `std::filesystem::`, then it will nicely gell with the existing apis. 

```c++
namespace nix::fs { using namespace std::filesystem; }
```

I do because if I do

```c++
using fs = std::filesystem;
```
then that `fs` will shadow, rather than expand, the `namespace fs` I put in `file-system.hh`.

I tried a few combinations, and the thing you commented on is the only one that will work (other than declaring `symlink_exists` in `std::filesystem` directly, which would be ill-advised.)

Reply:
clangd reports for an `fs::current_path` identifier:

> function current_path
> provided by <filesystem>
>
>→ path
>
> // In namespace std::filesystem
> path current_path()

This pattern also seems useful for polyfilling functions from new C++ versions.